onnxruntime/core/codegen/common/registry.h

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+#include "core/common/common.h"
+#include <functional>
+#include <string>
+#include <unordered_map>
+
+namespace onnxruntime {
+namespace codegen {
+
+// RegistryBase is a customized unordered_map
+// that keep ownership of passes,
+// including 1) IR builder passes
+//           2) Weight layout transformer passes
+//           3) Scheduler passses, etc.
+
```

Comment:
@KeZhang, many components under this "codegen/common" folder, like registry/profile/dispatcher, seems should be the task-transparent component that could shared with runtime. I feel it is a little weird if we have a "registry" type in runtime codebase but not used in other places.

Reply:
We had some discussion about this before. For the sake of minimize change to framework, these utils are kept in codegen/common.

---
In reply to: [294414368](https://github.com/microsoft/onnxruntime/pull/888#discussion_r294414368) [](ancestors = 294414368)